tests/forte/processors/base/data_augment_replacement_processor_test.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for data augment processors
+"""
+
+import unittest
+import tempfile
+import os
+from ddt import ddt, data, unpack
+from ft.onto.base_ontology import Token, Sentence, Document, Annotation
+from forte.processors.data_augment.algorithms.text_replacement_op import TextReplacementOp
+from forte.data.selector import AllPackSelector
+from forte.data.span import Span
+from forte.pipeline import Pipeline
+from forte.data.multi_pack import MultiPack
+from forte.data.ontology.top import MultiPackLink, MultiPackGroup, Link, Group
+from forte.data.readers import MultiPackSentenceReader, StringReader
+from forte.data.caster import MultiPackBoxer
+from forte.processors.base.data_augment_processor import ReplacementDataAugmentProcessor, modify_index
+from forte.processors.nltk_processors import NLTKWordTokenizer, NLTKPOSTagger
+
+
+class TmpReplacer(TextReplacementOp):
+    def __init__(self, configs={}):
+        super().__init__(configs)
+        self.token_replacement = {
+            "Mary": "Virgin",
+            "noon": "12",
+            "station": "stop",
+            "until": "til",
+        }
+
+    def replace(self, input):
+        return True, self.token_replacement.get(input.text, input.text)
+
+
```

Comment:
Thanks for adding more tests on the processor side.

In the codecov, I found that several `False` conditions are not reached. Take a look here: https://codecov.io/gh/asyml/forte/pull/285/src/forte/processors/base/data_augment_processor.py?before=forte/processors/base/data_augment_processor.py

Such as line 331, line 675.

Let me know what you think. We are a bit more cautious on the test cases here because you know that this function is a bit more complex than usual classes.

Reply:
Note that for some of these, it might be difficult to create test cases from the processor level, but it is ok to have some unit test only, that may allow the program to fall into one of the branch.